File: src/main/java/hr/fer/infusus/projekt/domain/Dogadaj.java
package hr.fer.infusus.projekt.domain;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Set;

@Data
public class Dogadaj {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column
    private String naziv;

    @Column
    private Integer trajanjeDani;

    @Column
    private Integer brojUlaznica;

    @Column
    private LocalDate datumPocetka;

    @JoinColumn(name = "mjesto_id")
    @ManyToOne
    private Mjesto mjesto;

    @Column
    @ManyToMany
    private Set<Izvodac> izvodaci;

    @Column
    @ManyToMany
    private Set<Menadzer> organizatori;
}

File: festival-frontend/src/app/shared/model/dogadaj.ts
import {Mjesto} from "./mjesto";
import {Izvodac} from "./izvodac";
import {Menadzer} from "./menadzer";

export interface Dogadaj {
  id: number;
  naziv: string;
  trajanjeDani: number;
  brojUlaznica: number;
  datumPocetka: any;
  mjesto: Mjesto;
  izvodaci: Izvodac;
  organizatori: Menadzer;
}
